module/private_quarters/private_quarters.py

```diff
@@ -0,0 +1,289 @@
+
+class PrivateQuarters(UI):
+    # Key: str, target ship name
+    # Value: Button, button instance
+    available_targets = {
+        'anchorage': PRIVATE_QUARTERS_SHIP_ANCHORAGE,
+        'noshiro': PRIVATE_QUARTERS_SHIP_NOSHIRO,
+        'sirius': PRIVATE_QUARTERS_SHIP_SIRIUS,
+    }
+
+    def _pq_target_appear(self):
+        """
+        Callable wrapper to validate target's appearance
+        offset=(100, 100) detectable for anchorage, noshiro, and sirus
+        When more ships added may need to adjust or capture specific bubble position per
+        ship, can use the available_targets to store similarly into tuples instead
+        """
+        settle_timer = Timer(1.5, count=3).start()
+        settle_timer.wait()
+        self.device.screenshot()
+        return self.appear(PRIVATE_QUARTERS_ROOM_TARGET_CHECK, offset=(100, 100))
+
+    def _pq_goto_room_check(self):
+        """
+        Callable wrapper for whether is loading or blocked by download asset popup
+        """
+        return self.appear(PRIVATE_QUARTERS_LOADING_CHECK, offset=(20,20)) or self.appear(POPUP_CANCEL, offset=(20, 20))
+
+    def _pq_goto_room_enter(self, target_ship):
+        """
+        Execute enter room routine
+
+        Args:
+            target_ship (str):
+
+        Returns:
+            bool
+        """
+        # Initiate goto into target's room
+        # Ensure either loading or popup
+        # prompt appears after click
+        target_title = target_ship.title()
+        target_btn = self.available_targets[target_ship]
+        self.ui_click(
+            click_button=target_btn,
+            check_button=self._pq_goto_room_check,
+            appear_button=page_private_quarters.check_button,
+            offset=(20, 20),
+            skip_first_screenshot=False)
+
+        # If was download asset popup
+        # Terminate the run
+        if self.handle_popup_cancel('PRIVATE_QUARTERS_DOWNLOAD_ASSET', offset=(20, 20)):
+            logger.error(f'Cannot enter {target_title}\'s room, please download the necessary assets first')
+            return False
+
+        # Fully enter into target's room
+        # through click progression
+        click_timer = Timer(1.5, count=3).start()
+        skip_first_screenshot = True
+        while 1:
+            if skip_first_screenshot:
+                skip_first_screenshot = False
+            else:
+                self.device.screenshot()
+
+            # End
+            if self.appear(PRIVATE_QUARTERS_ROOM_CHECK, offset=(20, 20)):
+                break
+
+            # Continue without clicking, mitigate too many click exception
+            if self.appear(PRIVATE_QUARTERS_LOADING_CHECK, offset=(20, 20)):
+                continue
+
+            if click_timer.reached():
+                self.device.click(PRIVATE_QUARTERS_ROOM_SAFE_CLICK_AREA)
+                click_timer.reset()
```

Comment:
clicking need to be triggered by `appear(button)`

Reply:
after loading, the ship girl will talk and there aren't any UI elements to really verify or validate on so have to rely on clicking a random space to progress the conversation and fully enter the room

the ship girl themselves is 3d so can't really count on a static image from it

Reply:
following up, after the loading screen besides the ship girl is the only UI element in the shot besides her personal room belongings

But having to capture each girl's specific room doesn't seem productive when the idea is to progress forward to get to the actual UI of the private quarters feature.